src/context.rs

```diff
@@ -23,6 +23,7 @@ pub trait ProviderExt<'a>: Provider<'a> {
 }
 impl<'a, P: Provider<'a>> ProviderExt<'a> for P {}
 
+#[allow(clippy::exhaustive_structs)]
```

Comment:
I guess the `Empty` is part of the API, and it will be empty. Otherwise we may want `#[non_exhaustive]` and provide construct method.

Reply:
I think it's fine to leave `EmptyProvider` as an exhaustive struct given that it'll never contain any data and is just here to provide an Provider for the edgecases where you need one but have no data you could possible provide. But I think I'll add a `new` method anyway given that it's usually expected to exist in Rust